src/components/Plugin/DaoModule/TransactionPreview.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <b
+    v-text="`${transaction.operation === '0' ? 'Call' : 'Delegate Call'} to`"
+  /><br />
+  <span>
+    <jazzicon
+      :address="transaction.to"
+      :diameter="22"
+      class="d-inline-block v-align-middle line-height-0"
+    />
+    &nbsp;<b v-text="_shorten(transaction.to)" />&nbsp;
+    <a
+      @click="copyToClipboard('transaction target', transaction.to)"
+      class="text-gray"
+      style="padding-top: 2px;"
+    >
+      <Icon name="copy" size="22" />
+    </a>
+    <br />
+  </span>
+  with<br />
+  <span
+    class="mr-1 tooltipped tooltipped-multiline tooltipped-n break-word"
+    :aria-label="formatAmount(transaction.value)"
+  >
+    {{ formatAmount(transaction.value, 5) }}
+  </span>
+  <br />
+  <span v-if="transaction.data">
+    and<br />
+    <span
+      class="mr-1 tooltipped tooltipped-multiline tooltipped-n break-word"
+      :aria-label="transaction.data"
+    >
+      {{ _shorten(transaction.data, 10) }}
+    </span>
+    <a
+      @click="copyToClipboard('transaction data', transaction.data)"
+      class="text-gray"
+      style="padding-top: 2px;"
+    >
+      <Icon name="copy" size="22" />
+    </a>
+  </span>
+</template>
+
+<script>
+import { formatEther } from '@ethersproject/units';
+export default {
+  props: ['transaction'],
+  async created() {
+    console.log('TransactionPReview');
+  },
+  methods: {
+    formatAmount(amount, maxDecimals) {
+      let out = formatEther(amount);
+      if (maxDecimals && out.includes('.')) {
+        const parts = out.split('.');
+        if (parts[1].length > maxDecimals) {
+          out = '~' + parts[0] + '.' + parts[1].slice(0, maxDecimals);
+        }
+      }
+      return out + ' ETH';
+    },
+    copyToClipboard(label, data) {
```

Comment:
We've already integrated a way to copy content, it would be better to use the same way to reduce code, have a look here for an example: https://github.com/snapshot-labs/snapshot/blob/218980f9b968de1eff997c983fd250543347a152/src/views/Settings.vue#L23-L29